app/src/main/kotlin/org/koitharu/kotatsu/list/ui/model/ListHeader.kt

```diff
@@ -4,32 +4,14 @@ import android.content.Context
 import androidx.annotation.StringRes
 import org.koitharu.kotatsu.core.ui.model.DateTimeAgo
 
-class ListHeader private constructor(
-	private val text: CharSequence?,
-	@StringRes private val textRes: Int,
-	private val dateTimeAgo: DateTimeAgo?,
-	@StringRes val buttonTextRes: Int,
-	val payload: Any?,
+data class ListHeader(
```

Comment:
We avoid using data classes whenever possible on purpose. For data class compiler generate also `componentN()` and `copy` functions which are not needed but might increase apk size. Long story short, we use data classes only if we need `copy` function

Reply:
This constructor also allow to pass, for instance, both `text` and `textRes`, or do not pass any arguments, that will produce invalid state

Reply:
The extra functions are removed by R8 during code shrinking if they aren't used.

Reply:
Hmm, I guess it makes sense

Reply:
> This constructor also allow to pass, for instance, both `text` and `textRes`, or do not pass any arguments, that will produce invalid state

In that case, I'll remove the default arguments.

Reply:
I'll try to replace some classes with `equals`/`hashCode` with `data`  and compare apk size